### DISCUSSION TITLE
ES Indices rotation

### DISCUSSION THREAD

**Discussion by tuamortemo** [Security-Onion-Solutions/securityonion#13465] [2.4] [answered]
### Version

2.4.90

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

16-32

### RAM

16-32

### Storage for /

200-300 GB

### Storage for /nsm

1,7 TB

### Network Traffic Collection

span port

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Good day. Question about index rotation in elasticsearch. The so-elasticsearch-indices-delete-delete script works strangely. Sometimes when nsm on search servers and manager server reaches >85%, judging by the logs /opt/so/log/so-elasticsearch-indices-delete.log, the script starts and starts deleting old indexes. But in most cases this does not happen, which is why the processing of new events stops. This often happens when servers receive a kernel update and are in the reboot state. The essence of the question: does the so-elasticsearch-indices-delete script depend on the overall fill rate of /nsm on servers with ES or does it focus only on the storage and deletion time, which is specified in the settings? Is it possible to edit it so that deletion starts at 79% of the /nsm load?

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by TOoSmOotH** [comment, marked as answer]
This script is really a last resort for users who do not tune ILM. Once you have properly tuned your ILM its advised to disable this delete script. 

https://docs.securityonion.net/en/2.4/elasticsearch.html#index-management
